Src/Support/IntegrationTests/ExternalAccountCredentialTests.cs

```diff
@@ -0,0 +1,115 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+using Google.Apis.Auth.OAuth2;
+using Google.Apis.IAMCredentials.v1;
+using Google.Apis.IAMCredentials.v1.Data;
+using Google.Apis.Services;
+using Google.Apis.Storage.v1;
+using Google.Apis.Storage.v1.Data;
+using IntegrationTests.Utils;
+using System;
+using System.Net.Http;
+using System.Threading.Tasks;
+using System.Xml;
+using Xunit;
+
+namespace IntegrationTests
+{
+    public class ExternalAccountCredentialTests
+    {
+        [Fact(Skip = "Googlers see go/wif-integration-settings")]
+        public async Task AwsCredentials()
+        {
+            string gcpProjectNumber = "";
+            string gcpPoolId = "pool-abc123d4";
+
+            string awsAccountId = "";
+            string awsProviderId = "aws-abc123d4";
+            string awsRoleName = "";
+            string awsAudience = $"//iam.googleapis.com/projects/{gcpProjectNumber}/locations/global/workloadIdentityPools/{gcpPoolId}/providers/{awsProviderId}";
+            string awsArn = $"arn:aws:iam::{awsAccountId}:role/{awsRoleName}";
+
+            // This really won't be running in AWS, so we need to obtain AWS credentials by using the AssumeRoleWithWebIdentity
+            // operation of AWS STS server.
+            // For this, we need an identity token to identify ourselves to AWS STS.
+            GoogleCredential gcpCredential = Helper.GetServiceCredential().CreateScoped("https://www.googleapis.com/auth/cloud-platform");
+            string saEmail = (gcpCredential.UnderlyingCredential as ServiceAccountCredential).Id;
+            string saResourceName = $"projects/-/serviceAccounts/{saEmail}";
+            IAMCredentialsService iamService = new IAMCredentialsService(new BaseClientService.Initializer
+            {
+                HttpClientInitializer = gcpCredential,
+            });
+            var idToken = await iamService.Projects.ServiceAccounts.GenerateIdToken(
+                new GenerateIdTokenRequest
+                {
+                    Audience = awsAudience,
+                    IncludeEmail = true
+                },
+                saResourceName).ExecuteAsync();
+
+            // Using the id token, we can now obtain AWS credentials.
+            string assumeRoleUrl = $"https://sts.amazonaws.com/?" +
+                $"Action=AssumeRoleWithWebIdentity" +
+                $"&Version=2011-06-15" +
+                $"&DurationSeconds=3600" +
+                $"&RoleSessionName={awsRoleName}" +
+                $"&RoleArn={awsArn}" +
+                $"&WebIdentityToken={idToken.Token}";
+
+            string content;
+            using (HttpClient httpClient = new HttpClient())
+            {
+                HttpResponseMessage response = await httpClient.GetAsync(assumeRoleUrl);
+                content = await response.Content.ReadAsStringAsync();
+            }
+            XmlDocument awsCredentialXml = new XmlDocument();
```

Comment:
I'd suggest using LINQ to XML instead of XmlDocument in general - it's a much nicer API.

Reply:
Yep done. It was very simple what we needed to do, but in case that changes in the future, it's better with Linq.